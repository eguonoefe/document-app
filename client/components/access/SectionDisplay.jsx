import React from 'react';
import { Link } from 'react-router-dom';
/**
 * Section Display component that displays a carousel.
 * @returns {String} The HTML markup for the Section Display component
 */
const SectionDisplay = () => (
  <div className="section">
    <div className="carousel carousel-slider center" data-indicators="true">
      <div className="carousel-fixed-item">
        <span className="inner-slide">
          <img
            src="../../img/slide5.jpg"
            alt="slide1" className="slide-img" />
        </span>
        <div className="landing-content">
          <h2>DocNinja</h2>
          <Link to="/signup">Get Started</Link>
        </div>
      </div>
      <div className="carousel-item white-text" href="#one!">
        <img
          src="../../img/slide6.jpg"
          alt="slide" className="slide-img" />
      </div>
      <div className="carousel-item amber white-text" href="#two!">
        <img
          src="../../img/slide1.jpg"
          alt="slide3" className="slide-img" />
      </div>
      <div className="carousel-item green white-text" href="#three!">
        <img
          src="../../img/slide4.jpg"
          alt="slide4" className="slide-img" />
      </div>
      <div className="carousel-item blue white-text" href="#four!">
        <img
          src="../../img/slide3.jpg"
          alt="slide3" className="slide-img" />
      </div>
    </div>
  </div>
);

export default SectionDisplay;
