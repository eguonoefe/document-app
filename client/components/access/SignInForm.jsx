import React, { useState, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * SignIn component for logging in a user
 * @param {Object} props { logInUser }
 * @returns {String} The HTML markup for the SignInForm
 */
const SignInForm = ({ logInUser }) => {

  const [values, setValues] = useState({email: '', password: ''});

  const onChange = e => {
    const {name, value} = e.target;
    setValues({ ...values, [name]: value });
  }

  const submit = () => logInUser(values)

  return (
    <Fragment>
      <h5 className="center"> Log Into Doc-ms</h5>
      <div className="input-field">
        <input
          className="validate form-design"
          type="email"
          name="email"
          value={values.email}
          onChange={onChange}
          placeholder="Email"
          required
        />
      </div>
      <div className="input-field">
        <input
          className="validate form-design"
          type="password"
          name="password"
          value={values.password}
          onChange={onChange}
          placeholder="password"
          required
        />
      </div>
      <div className="input-field center">
        <button
          id="submit-signin"
          className="waves-effect btn button-design"
          onClick={submit}
        >
          Sign In
        </button>
      </div>
      <p className="center">
        Don&#39;t Have an account? <Link to="signup">Create an Account</Link>
      </p>
    </Fragment>
  );
};

SignInForm.propTypes = {
  logInUser: PropTypes.func.isRequired
};

export default SignInForm;
