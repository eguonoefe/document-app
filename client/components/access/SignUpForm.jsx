import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';

import AccessActions from '../../actions/AccessActions';
import validate from '../../utilities/validate';


const signUpUser = AccessActions.signUpUser;
/**
 * @class SignUpForm
 * @extends {Component}
 */
export class SignUpForm extends Component {
  /**
   * Creates an instance of SignUpForm.
   * @param {Object} props -
   * @memberof SignUpForm
   */
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      confirmPassword: ''
    };
    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  /**
  * Validates the SignUpForm
  * Makes an action call to sign up the user
  * Toasts the error/success message
  * @return {void}
  * @memberof SignUpForm
  */
  onSubmit() {
    try {
      const { valid } = validate.validateSignUp(this.state);
      if (!valid) {
        throw new Error('No field should be left blank');
      }
      this.props.signUpUser(this.state)
        .then(() => {
          if (this.props.access.message) {
            return Materialize.toast(this.props.access.message,
              2000, 'indigo darken-4 white-text rounded');
          }
          Materialize.toast('Welcome!',
            2000, 'indigo darken-4 white-text rounded');
          this.props.history.push('/dashboard');
        });
    } catch (err) {
      Materialize.toast(err.message, 3000,
        'indigo darken-4 white-text rounded');
    }
  }

  /**
  * Sets the event value to the state
  * @return {void}
  * @param {Object} event The event of the HTML component
  * @memberof SignUpForm
  */
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  /**
  * Renders the SignUpForm component
  * @returns {String} The HTML markup for the SignUpForm
  * @memberof SignUpForm
  */
  render() {
    return (
      <div>
        <h5 className="center"> Create an account </h5>
        <div className="row">
          <div className="col l6 m6 s12">
            <div className="input-field">
              <input
                className="validate form-design"
                name="firstName"
                value={this.state.firstName}
                placeholder="First Name"
                type="text" onChange={this.onChange}
                required
              />
            </div>
          </div>
          <div className="col l6 m6 s12">
            <div className="input-field">
              <input
                name="lastName"
                value={this.state.lastName}
                placeholder="Last Name"
                className="validate form-design"
                type="text" onChange={this.onChange}
                required
              />
            </div>
          </div>
          <div className="col l12 m12 s12">
            <div className="input-field">
              <input
                name="email"
                value={this.state.email}
                placeholder="Email"
                className="validate form-design"
                type="email" onChange={this.onChange}
                required
              />
            </div>
          </div>
          <div className="col l12 m12 s12">
            <div className="input-field">
              <input
                name="password"
                value={this.state.password}
                placeholder="Password"
                className="validate form-design"
                type="password" onChange={this.onChange}
                required
              />
            </div>
          </div>
          <div className="col l12 m12 s12">
            <div className="input-field">
              <input
                name="confirmPassword"
                value={this.state.confirmPassword}
                placeholder="Re-Type Password"
                className="validate form-design"
                type="password" onChange={this.onChange}
                required
              />
            </div>
          </div>
        </div>
        <div className="input-field center">
          <button
            className="waves-effect btn button-design" id="signup-button"
            onClick={this.onSubmit}>
            Sign Up
          </button>
        </div>
        <p className="center">Have an account? <Link to="signin">Sign In</Link>
        </p>
      </div >
    );
  }
}

const mapPropsToState = state => (
  {
    access: state.access
  }
);

SignUpForm.propTypes = {
  signUpUser: PropTypes.func,
  access: PropTypes.object.isRequired,
  history: PropTypes.object
};


export default connect(mapPropsToState, { signUpUser })(withRouter(SignUpForm));
