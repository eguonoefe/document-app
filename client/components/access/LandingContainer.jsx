import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ConnectedHeader from '../include/Header';
import SectionDisplay from './SectionDisplay';

/**
 * @class LandingContainer
 * @extends {Component}
 */
export class LandingContainer extends Component {

  /**
   * Initializes the carousel component
   * Redirects user to dashbaord page if authenticated
   * @return {void}
   * @memberof LandingContainer
   */
  componentDidMount() {
    $('.carousel.carousel-slider').carousel({ fullWidth: true });
    setInterval(() => {
      $('.carousel').carousel('next');
    }, 7000);
    if (this.props.access.isAuthenticated === true) {
      this.props.history.push('/dashboard');
    }
  }

  /**
   * Renders a component
   * @returns {void}
   * @memberof LandingContainer
   */
  render() {
    return (
      <div className="landing-page">
        <ConnectedHeader match={this.props.match} />
        <SectionDisplay />
      </div>
    );
  }
}

const mapPropsToState = state => (
  {
    access: state.access
  }
);

LandingContainer.propTypes = {
  access: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
  history: PropTypes.object
};


export default connect(mapPropsToState)(withRouter(LandingContainer));
