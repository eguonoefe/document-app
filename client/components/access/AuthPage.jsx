import React, { Component } from "react";
import { Switch, Route, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import ConnectedHeader from "../include/Header";

import SignInForm from "./SignInForm";
import SignUpConnectedForm from "./SignUpForm";

import AccessActions from "../../actions/AccessActions";
import validate from "../../utilities/validate";

const signInUser = AccessActions.signInUser;

/**
 * @class LandingContainer
 * @extends {Component}
 */
export class AuthPage extends Component {
  constructor(props) {
    super(props);

    this.logInUser = this.logInUser.bind(this);
  }
  /**
   * Validates the log in user details
   * Makes an action call to sign in the user
   * Toasts the error/success message
   * @return {void}
   * @memberof AuthPage
   */
  async logInUser(loginDetails) {
    try {
      const { valid } = validate.validateLogin(loginDetails);
      if (!valid) throw new Error("No field should be left blank");

      const user = await this.props.signInUser(loginDetails);
      if (this.props.access.message) {
        return Materialize.toast( this.props.access.message, 2000, "indigo darken-4 white-text rounded");
      }

      Materialize.toast("Welcome!", 2000, "indigo darken-4 white-text rounded");
      this.props.history.push("/dashboard");
    } catch (err) {
      Materialize.toast( err.message, 3000, "indigo darken-4 white-text rounded");
    }
  }

  /**
   * Renders a component
   * @returns {void}
   * @memberof LandingContainer
   */
  render() {
    return (
      <div>
        <ConnectedHeader match={this.props.match} />
        <div className="container authpage">
          <div className="row auth-card">
            <Switch>
              <Route
                exact
                path={`${this.props.match.path}/signin`}
                render={props => (
                  <SignInForm {...props} logInUser={this.logInUser} />
                )}
              />
              <Route
                exact
                path={`${this.props.match.path}/signup`}
                component={SignUpConnectedForm}
              />
            </Switch>
          </div>
        </div>
      </div>
    );
  }
}
const mapPropsToState = state => ({
  access: state.access
});

AuthPage.propTypes = {
  match: PropTypes.object.isRequired,
  history: PropTypes.object
};

export default connect(mapPropsToState, { signInUser })(withRouter(AuthPage));
