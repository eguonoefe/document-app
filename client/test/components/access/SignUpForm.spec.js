/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { SignUpForm } from '../../../components/access/SignUpForm';

configure({ adapter: new Adapter() });

describe('SignUpForm Component', () => {
  const spy = jest.fn();
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  afterEach(() => {
    spy.mockReset();
  });
  const props = {
    match: {
      path: ''
    },
    access: {
      isAuthenticated: false
    },
    history: {
      push: spy
    },
    signUpUser: jest.fn(() => {
      return Promise.resolve();
    })
  };
  const component = shallow(
    <SignUpForm {...props} />);
  test('should match the SignUpForm snapshot', () => {
    const tree = toJson(component);
    expect(component.find('input').length).toEqual(5);
    expect(tree).toMatchSnapshot();
  });
  test('it should set state when onChange function is called', () => {
    component.instance().onChange(
      { target: { value: 'efe@gmail.com', name: 'email' } });
    component.instance().onChange(
      { target: { value: 'eguono', name: 'password' } });
    component.instance().onChange(
      { target: { value: 'eguono', name: 'firstName' } });
    component.instance().onChange(
      { target: { value: 'eguono', name: 'lastName' } });
    component.instance().onChange(
      { target: { value: 'eguono', name: 'confirmPassword' } });
    expect(component.state('email')).toEqual('efe@gmail.com');
    expect(component.state('password')).toEqual('eguono');
    expect(component.state('firstName')).toEqual('eguono');
    expect(component.state('lastName')).toEqual('eguono');
    expect(component.state('confirmPassword')).toEqual('eguono');
    const newspy = jest.spyOn(component.instance(), 'onSubmit');
    component.instance().onSubmit();
    expect(newspy).toHaveBeenCalled();
  });
  test(
    'it should submit fields in state when onSubmit function is called', () => {
      jest.spyOn(component.instance(), 'onSubmit');
      component.instance().onChange(
        { target: { value: 'efe@gmail.com', name: 'email' } });
      component.instance().onChange(
        { target: { value: 'eguono', name: 'password' } });
      component.find('#signup-button').simulate('click');
      expect(component.find('#signup-button').length).toEqual(1);
      expect(component.instance().onSubmit.mock.calls.length).toEqual(2);
    });
});
