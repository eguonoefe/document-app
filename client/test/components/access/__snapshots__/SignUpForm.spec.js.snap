// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm Component should match the SignUpForm snapshot 1`] = `
<div>
  <h5
    className="center"
  >
     Create an account 
  </h5>
  <div
    className="row"
  >
    <div
      className="col l6 m6 s12"
    >
      <div
        className="input-field"
      >
        <input
          className="validate form-design"
          name="firstName"
          onChange={[Function]}
          placeholder="First Name"
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="col l6 m6 s12"
    >
      <div
        className="input-field"
      >
        <input
          className="validate form-design"
          name="lastName"
          onChange={[Function]}
          placeholder="Last Name"
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="col l12 m12 s12"
    >
      <div
        className="input-field"
      >
        <input
          className="validate form-design"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      className="col l12 m12 s12"
    >
      <div
        className="input-field"
      >
        <input
          className="validate form-design"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="col l12 m12 s12"
    >
      <div
        className="input-field"
      >
        <input
          className="validate form-design"
          name="confirmPassword"
          onChange={[Function]}
          placeholder="Re-Type Password"
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="input-field center"
  >
    <button
      className="waves-effect btn button-design"
      id="signup-button"
      onClick={[Function]}
    >
      Sign Up
    </button>
  </div>
  <p
    className="center"
  >
    Have an account? 
    <Link
      to="signin"
    >
      Sign In
    </Link>
  </p>
</div>
`;
